src/date.c

```diff
@@ -1,18 +1,1012 @@
+    d_argtoint(v7, &arg0, &ret_time);
+  }
+
+  n = v7_create_number(ret_time);
+  v7_set_property(v7, this_obj, "", 0, V7_PROPERTY_HIDDEN, n);
+  return n;
+}
+
+
+#define DEF_SET_TP(name, start_pos) \
+  static val_t Date_setUTC##name(struct v7 *v7, val_t this_obj, val_t args) { \
+    return d_setTimePart(v7, this_obj, args, start_pos, d_gmtime, d_gmktime); \
+  }\
+  static val_t Date_set##name(struct v7 *v7, val_t this_obj, val_t args) { \
+    return d_setTimePart(v7, this_obj, args, start_pos, \
+                                          d_localtime, d_lmktime); \
+  }
+
+DEF_SET_TP(Milliseconds, tpmsec)
+DEF_SET_TP(Seconds, tpseconds)
+DEF_SET_TP(Minutes, tpminutes)
+DEF_SET_TP(Hours, tphours)
+DEF_SET_TP(Date, tpdate)
+DEF_SET_TP(Month, tpmonth)
+DEF_SET_TP(FullYear, tpyear)
+
+/*++++ other API ++++*/
+
+static val_t Date_toJSON(struct v7 *v7, val_t this_obj, val_t args) {
+  return Date_toISOString(v7, this_obj, args);
+}
+
+static val_t Date_valueOf(struct v7 *v7, val_t this_obj, val_t args) {
+  (void)args;
+  if(!v7_is_object(this_obj) || (v7_is_object(this_obj)
+     && v7_to_object(this_obj)->prototype != v7_to_object(v7->date_prototype))){
+    throw_exception(v7, "TypeError", "Date.valueOf called on non-Date object");
+  }
+  
+  return Obj_valueOf(v7, this_obj, args);
+}
+
+static val_t Date_getTime(struct v7 *v7, val_t this_obj, val_t args) {
+  return Date_valueOf(v7, this_obj, args);
+}
+
+static val_t Date_getTimezoneOffset(struct v7 *v7, val_t this_obj, val_t args) {
+  (void)args; (void)v7; (void)this_obj;
+  return v7_create_number(gmtoffms/msPerMinute);
+}
+
+static val_t Date_now(struct v7 *v7, val_t this_obj, val_t args) {
+  etime_t ret_time;
+  (void)args;
+  
+  if(!d_iscalledasfunction(v7, this_obj)) {
+    throw_exception(v7, "TypeError", "Date.now() called on object");
+  }
+
+  d_gettime(&ret_time);
+  
+  return v7_create_number(ret_time);
+}
+
+static val_t Date_parse(struct v7 *v7, val_t this_obj, val_t args) {
+  etime_t ret_time = INVALID_TIME;
+  
+  (void)args;
+  
+  if(!d_iscalledasfunction(v7, this_obj)) {
+    throw_exception(v7, "TypeError", "Date.parse() called on object");
+  }
+  
+  if(v7_array_length(v7, args) >= 1) {
+    val_t arg0 = v7_array_at(v7, args, 0);
+    if(v7_is_string(arg0)) {
+      size_t size;
+      const char* time_str = v7_to_string(v7, &arg0, &size);
+      
+      d_timeFromString(&ret_time, time_str, size);
+    }
+  }
+  
+  return v7_create_number(ret_time);
+}
+
+static val_t Date_UTC(struct v7 *v7, val_t this_obj, val_t args) {
+  etime_t ret_time;
+  (void)args;
+  
+  if(!d_iscalledasfunction(v7, this_obj)) {
+    throw_exception(v7, "TypeError", "Date.now() called on object");
+  }
+
+  ret_time = d_time_number_from_tp(v7, this_obj, args, tpyear, 0, d_gmktime);
+  return v7_create_number(ret_time);
+}
+
+/****** Initialization *******/
+
+/* we should set V7_PROPERTY_DONT_ENUM for all Date props */
+/* TODO(???): check other objects */
```

Comment:
Probably it's counterintuitive, but we tend to put always the author of the comment in the `TODO()` mark. It's not the person who has to do it, but rather the person who figured out that there is something to do. The idea is that that person is likely to know more about what and why has to be done.